src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,58 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int STORAGE_MAX_CAPACITY = 10;
+    private int size;
+    private final StorageElement<K, V>[] storageElements = new StorageElement[STORAGE_MAX_CAPACITY];
+
     @Override
     public void put(K key, V value) {
+
+        for (int i = 0; i < STORAGE_MAX_CAPACITY; i++) {
+            if (storageElements[i] == null) {
+                storageElements[i] = new StorageElement<>(key, value);
+                size++;
+                return;
+            } else if (storageElements[i].isEqualToKey(key)) {
+                storageElements[i].value = value;
+                return;
+            }
+        }
+        throw new IllegalStateException("StorageImpl has no more space. Max space is"
+                + STORAGE_MAX_CAPACITY);
```

Comment:
would be more handy to have a size check as a separate if, after the value replacement logic but before the new key logic, so that the exception that you throw would be close to the reason why it is thrown thus improving readability
```suggestion
        for (int i = 0; i < size; i++) {
             if (storageElements[i].isEqualToKey(key)) {
                storageElements[i].value = value;
                return;
            }
        }
        if (size >= STORAGE_MAX_CAPACITY) {
            throw new IllegalStateException("StorageImpl has no more space. Max space is: "
                + STORAGE_MAX_CAPACITY);
        }
        storageElements[size++] = new StorageElement<>(key, value);
```